github/actions_artifacts.go

```diff
@@ -160,6 +168,26 @@ func (s *ActionsService) DownloadArtifact(ctx context.Context, owner, repo strin
 	return parsedURL, newResponse(resp), nil
 }
 
+func (s *ActionsService) downloadArtifactWithRateLimit(ctx context.Context, u string, maxRedirects int) (*url.URL, *Response, error) {
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	url, resp, err := s.client.bareDoUntilFound(ctx, req, maxRedirects)
+	if err != nil {
+		return nil, resp, err
+	}
+	defer resp.Body.Close()
+
+	// If we received a valid Location in a 302 response
+	if url != nil {
+		return url, resp, nil
+	}
+
+	return nil, resp, fmt.Errorf("unexpected status code: %s", resp.Status)
```

Comment:
For this file and all the other `...WithRateLimit` methods, could we place make the "happy path" the one that flows to the end of the method, as per idiomatic Go?

```suggestion
	// If we didn't receive a valid Location in a 302 response
	if url == nil {
		return nil, resp, fmt.Errorf("unexpected status code: %v", resp.Status)
	}

	return url, resp, nil
```